src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -29,8 +29,14 @@ public class Command
 
         private bool _running = false;
 
-        private Command(string executable, string args)
+        //the command is not loaded from nuget libraries or from AppContext.BaseDirectory. It is machine specific.
+        //see https://github.com/dotnet/cli/issues/382
+        private readonly bool _isUnsafe;
+
+        private Command(string executable, string args, bool isUnsafe)
         {
+            _isUnsafe = isUnsafe;
```

Comment:
It feels weird that Command (which is used by almost every `dotnet` tool) should know that loading from the path vs loading from nuget is unsafe.

It would make more sense if we exposed some public enum (or type of some kind) that specified how the command was loaded and compile can make a decision about the safety based on that loading mechanism.